.github/workflows/run-clang-format.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+# Copyright 2020 The Verible Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FORMAT_OUT=${TMPDIR:-/tmp}/clang-format-diff.out
+
+# TODO(hzeller): only run on the files that are checked out
+clang-format -i --style=Google $(find . -name "*.h" -o -name "*.cc")
```

Comment:
We could probably use something like:
`$(git ls-tree HEAD -r --name-only | grep --color=never -e '\.cc$\|\.h$')`

But assuming that this is run before building and before anything is generated I think `find` should be okay too

Reply:
There's a recipe for modified files here (`git_touched_files`):

https://cs.opensource.google/verible/verible/+/master:verilog/tools/formatter/verible-verilog-format-changed-lines-interactive.sh;drc=74b906bcefefbda3dea36abba4a6a2ce5707d5f0;l=127

Reply:
Both these assume that the files are local changed. Though I assume within the pull request environment it will already have applied the patch already as we have a clean head without diffs.
So maybe I need something like
```
git diff --name-only --diff-filter=AM HEAD~1
```
? Will have to try.

Reply:
So I did some experiments in a testing branch, trying to see [how the git repository looks like when we're called in an action in a pull request](https://github.com/hzeller/verible/blob/test-clang-format/.github/workflows/run-clang-format.sh#L18-L28)

It [looks weird](https://github.com/hzeller/verible/runs/1824649616#step:4:4): it looks like a git repository with a single branch (named like the branch from which is pushed) ,but no master branch. There is also only a single commit on it. So it looks like it  clones  everything, applies all the previous patches in the pull request, uses that as 'git init' baseline in a fresh repository with a single branch and then applies a single patch, which is the last push in that pull request.

Anyway, so this means, not even a `git diff --name-only --diff-filter=AM -r origin/master` will work.
Apparently, there are github actions scripts on the marketplace [that can find all changed files](https://github.com/marketplace/actions/get-all-changed-files); will have to look into that.

So I suggest for now, we leave it with the `find` all the files and apply clang-format to them. Which is fine, as we expect the code-base to be clean all the time so that we can afford running the format over all the files, even the ones not touched in the pull request.